src/platform/site-wide/mega-menu/index.js

```diff
@@ -9,14 +9,17 @@ import React from 'react';
 import { Provider } from 'react-redux';
 
 import startReactApp from '../../startup/react';
-import Main from './containers/Main';
 
 /**
  * Sets up the login widget with the given store at login-root
  *
  * @param {Redux.Store} store The common store used on the site
  */
-export default function startMegaMenuWidget(data, store) {
+export default async function startMegaMenuWidget(data, store) {
+  const {
+    default: Main,
+  } = await import(/* mega-menu-widget */ './containers/Main');
```

Comment:
Same here, this is for the header being used on desktops. Since the header is being used on every page, do we want to lazy load?